model/RepositoryService.php

```diff
@@ -90,10 +90,12 @@ public function getRevision($resourceId, $version)
      * (non-PHPdoc)
      * @see \oat\taoRevision\model\Repository::commit()
      */
-    public function commit($resourceId, $message, $version = null)
+    public function commit($resourceId, $message, $version = null, $userId = null)
     {
-        $user = \common_session_SessionManager::getSession()->getUser();
-        $userId = is_null($user) ? null : $user->getIdentifier();
+        if (!$userId) {
```

Comment:
I'm guessing user idea probably will never be '0', but just in case it might be better to make this a strict check `!$userId === null`

Reply:
Oh right; fixed!